src/main/java/uk/gov/hmcts/divorce/notification/pin/PinGenerationService.java

```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.divorce.notification.pin;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PinGenerationService {
+
+    private static final String ALLOWED_CHARS = "ABCDEFGJKLMNPRSTVWXYZ23456789";
+
+    public String generatePin() {
```

Comment:
unless you are using it at multiple places we can move it to `CitizenInviteApplicant2`